### PR TITLE
fix(DualList): Add item disability functionality.

### PR DESCRIPTION
**Pull request by Laviro**
When an item is disabled,
the user won't be able to move it between the lists:
dec 16 2018 4_22 pm

### CODE REVIEW

File: packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItem.js [hunk 1]
```diff
@@ -70,7 +92,10 @@ DualListItem.defaultProps = {
   filterTerm: null,
   onChange: noop,
   side: null,
-  hidden: false
+  hidden: false,
+  disabled: false,
+  tooltipID: `dual-list-item-tooltip-${UUID()}`,
```
Comment (dmiller9911) [inline packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItem.js]: This only runs once at module import. Not when the component is instantiated so if someone mounts multiple of these they will all share the same id.

Comment (Laviro) [reply packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItem.js]: Wow thanks.. I didn't know that..
maybe set the ID by the item side+position which is unique enough

Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItem.js]: What we've done in other components is to set the default to null then check for it and if it is not passed use:
```
const randomId = () => Date.now();
...
<Tooltip id={tooltipID || randomId()}>
```

Comment (Laviro) [reply packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItem.js]: Tell me what do you think, added this:
```
const getTooltipID = () => {
    let uniqueTooltipID = `dual-list-item-tooltip-${side}`;
    if (parentPosition) {
      uniqueTooltipID += `-${parentPosition}`;
    }
    uniqueTooltipID += `-${position}`;
    return uniqueTooltipID;
  };
```
and then using it like this when the default `tooltipID` is `null`.
```
    <DualListItemTooltip text={tooltipText} id={tooltipID || getTooltipID()}>
```

File: packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItemTooltip.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import UUID from 'uuid/v1';
```
[the remark this answers is not in the document]
Comment (Laviro) [reply packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItemTooltip.js]: oops, forgot to remove this one.. 
what do you think of the implementation above? isn't it unique enough ?
thanks

[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react/src/components/DualListSelector/components/DualListItemTooltip.js]: Yup. Thanks.

File: packages/patternfly-3/patternfly-react/less/dual-list-selector.less [hunk 3]
```diff
@@ -117,12 +117,22 @@
     color: white;
   }
 
+  &.disabled {
+    cursor: not-allowed;
+    background: @color-pf-black-150;
+    color: @color-pf-black-500;
```
Comment (dtaylor113) [inline packages/patternfly-3/patternfly-react/less/dual-list-selector.less]: nit. supposed to be in alphabetical order.

Comment (jeff-phillips-18) [reply packages/patternfly-3/patternfly-react/less/dual-list-selector.less]: we haven't been keeping up with that in this repo

**Pull request merged**